pennylane/qnode.py

```diff
@@ -667,6 +674,9 @@ def jacobian(self, params, which=None, *, method='B', h=1e-7, order=1, **kwargs)
             # construct the circuit
             self.construct(params, circuit_kwargs)
 
+        if any(e.return_type == 'sample' for e in self.ev):
+            raise QuantumFunctionError("Circuits that include sampling can not be differentiated.")
```

Comment:
For clarity, would be better to indicate exactly which op is non-differentiable here

Reply:
I think "don't use qml.sample anywhere if you want to differentiate your circuit" should be sufficient and the user will now where he used `qml.sample`.